File: Kernel Simulator/Misc/UserManual.vb

'    Kernel Simulator  Copyright (C) 2018-2019  EoflaOE
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Module UserManual

    'Variables
    Public sectionsManual As Dictionary(Of String, String)

    'This is not final! This is here only for preparation for 0.0.5.9's features.
    ''' <summary>
    ''' We haven't implemented anything yet.
    ''' </summary>
    ''' <param name="section">A section on the current manual page</param>
    ''' <remarks></remarks>
    Public Sub ReadManual(ByVal section As String)
        Throw New NotImplementedException
    End Sub

    ''' <summary>
    ''' We haven't implemented anything yet.
    ''' </summary>
    ''' <param name="sectionFile">A file which is a section file</param>
    ''' <remarks></remarks>
    Public Sub ParseManual(ByVal sectionFile As String)
        Throw New NotImplementedException
    End Sub

End Module

File: public/Nitrocid/Shell/Shells/UESH/Commands/UserManual.cs
//
// Nitrocid KS  Copyright (C) 2018-2025  Aptivi
//
// This file is part of Nitrocid KS
//
// Nitrocid KS is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Nitrocid KS is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY, without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

using Nitrocid.ConsoleBase.Colors;
using Nitrocid.ConsoleBase.Writers;
using Terminaux.Writer.ConsoleWriters;
using Nitrocid.Kernel;
using Nitrocid.Languages;
using Nitrocid.Shell.ShellBase.Commands;
using Textify.General;

namespace Nitrocid.Shell.Shells.UESH.Commands
{
    /// <summary>
    /// Shows the link to Nitrocid KS's wiki and its API doc for mods.
    /// </summary>
    class UserManualCommand : BaseCommand, ICommand
    {

        public override int Execute(CommandParameters parameters, ref string variableValue)
        {
            TextWriterColor.Write(Translate.DoTranslation("To access the user manual for general uses of Nitrocid, please visit the below link:"));
            TextWriters.Write(
                KernelPlatform.IsOnUnix() ?
                $"    {CharManager.GetEsc()}]8;;https://aptivi.gitbook.io/aptivi/nitrocid-ks-manual/{CharManager.GetEsc()}\\Manual page{CharManager.GetEsc()}]8;;{CharManager.GetEsc()}\\" :
                "    https://aptivi.gitbook.io/aptivi/nitrocid-ks-manual/",
                true, KernelColorType.Tip
            );
            TextWriterColor.Write(Translate.DoTranslation("To access the developer's manual for updated ways of making kernel modifications and screensavers, please visit the below link:"));
            TextWriters.Write(
                KernelPlatform.IsOnUnix() ?
                $"    {CharManager.GetEsc()}]8;;https://aptivi.github.io/Nitrocid{CharManager.GetEsc()}\\API documentation{CharManager.GetEsc()}]8;;{CharManager.GetEsc()}\\" :
                "    https://aptivi.github.io/Nitrocid",
                true, KernelColorType.Tip);
            return 0;
        }

        public override int ExecuteDumb(CommandParameters parameters, ref string variableValue)
        {
            TextWriterColor.Write(Translate.DoTranslation("To access the user manual for general uses of Nitrocid, please visit the below link:"));
            TextWriters.Write("    https://aptivi.gitbook.io/aptivi/nitrocid-ks-manual/", true, KernelColorType.Tip);
            TextWriterColor.Write(Translate.DoTranslation("To access the developer's manual for updated ways of making kernel modifications and screensavers, please visit the below link:"));
            TextWriters.Write("    https://aptivi.github.io/Nitrocid", true, KernelColorType.Tip);
            return 0;
        }

    }
}
